.github/workflows/go-unit-tests.yaml

```diff
@@ -0,0 +1,72 @@
+name: Unit Tests
+
+on:
+  push:
+    branches: ["main"]
+
+  pull_request:
+    branches: ["main"]
+
+jobs:
+  test:
+    name: Unit Tests
+    strategy:
+      matrix:
```

Comment:
Sorry for requests from the uninitiated, but once again I'm curious: What is matrix?

Reply:
All good: matrix strategies are super powerful, even though here we don't use them (yet).

Matrix strategies help you define various combinations, e.g. different go versions, which you can use to templatize the action.

Reason why I go with matrix setup even though it just uses one target (setting), is that later it's super easy to add another setting. For example, imagine you want to unit test various Go runtimes, or different go test settings. This all can be input to the matrix test and Github Actions will spin up multiple (concurrent) test pipelines based on the matrix table.

Reply:
OK, I get it, matrix is a keyword for this GitHub tool I haven't learned yet.  Another item for my to-do list.